heron/api/src/scala/com/twitter/heron/streamlet/scala/impl/BuilderImpl.scala

```diff
@@ -0,0 +1,41 @@
+//  Copyright 2018 Twitter. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+package com.twitter.heron.streamlet.scala.impl
+
+import com.twitter.heron.streamlet.scala.{Builder, Source, Streamlet}
+import com.twitter.heron.streamlet.scala.converter.ScalaToJavaConverter
+
+
+
+class BuilderImpl extends Builder {
+
+  override def newSource[R](supplierFn: () => R): Streamlet[R] = {
+    val serializableSupplier = ScalaToJavaConverter.toSerializableSupplier[R](supplierFn)
+    val newJavaStreamlet =
+      new com.twitter.heron.streamlet.impl.streamlets.SupplierStreamlet[R](
+        serializableSupplier)
+    StreamletImpl.toScalaStreamlet[R](newJavaStreamlet)
```

Comment:
This function name "toScalaStreamlet" seems to be a bit misleading. "fromJavaStreamlet" might be better?

Reply:
Will rename in a later commit

Reply:
@nwangtw - if the intention is to convert to - toScalaStreamlet - that will be fine.

Reply:
Yes that is indeed the intention

Reply:
Yep, this function transform `Java Streamlet` to `Scala Streamlet` version so why its naming convention is `toScalaSource`. Also, it is same convention what Scala currently has (i.e: `toList`, `toMap` etc..)